### ISSUE TITLE
Admin login and flash message does not update the page accordingly

### ISSUE THREAD

**Issue by milkolori** [magento/magento2#6533]
**System configuration:** 
1. Magento CE 2.1.1 without sample data is installed.
2. I am using a custom theme.
3. PHP version 7.0.10 is installed.
4. MySql version 5.6.30. is installed.
5. HTTPS protocol is used.

**Login issue**

_Steps to reproduce_
1. Navigate https://mymagentosite.com/admin
2. Enter the username and password
3. Click in the login button
4. The page is redirected back to the login screen without any message
5. Enter the username and password for second time
6. The page is redirected to admin dashboard

_Expected behaviour_
1. Navigate https://mymagentosite.com/admin
2. Enter the username and password
3. The page is redirected to admin dashboard

**Issue with displaying flash messages**

_Steps to reproduce_
1. Open System-> Cache Management from admin module
2. Click on the button "Flush Magento Cache"
3. The page is redirected to Cache Management screen without displaying any message
4. Navigate to dashboard
5. Dashboard page is loaded and message "The Magento cache storage has been flushed."

_Expected behaviour_
1. Open System-> Cache Management from admin module
2. Click on the button "Flush Magento Cache"
3. Redirect the page to Cache Management screen and display message "The Magento cache storage has been flushed."

**Additional information**
The problem started happening after I updated from magento 2.0.4 to magento 2.1.1. I used the composer to update magento version. After updating I execute the following commands in the console:

```
magento setup:upgrade
magento setup:di:compile
magento setup:static-content:deploy
```

**Comment by andimov** [comment]
@milkolori 
Thanks for reporting this issue.
Please use our issue reporting template to describe this issue. The template specifies the version and configuration information we need to assess your issue. See the [Issue reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines) for more information.

**Comment by milkolori** [comment]
@andimov 
Thank you for the notes. I did updat issue details.

**Comment by oshmyheliuk** [comment]
@milkolori
Can't reproduce this issue with information that you provided.
Please add additional info about operating system and browser that you used.

**Comment by milkolori** [comment]
@oshmyheliuk  
Here are more details about my environment:

**Server OS:** Linux version 2.6.32-673.26.1.lve1.4.15.el6.x86_64 (mockbuild@build.cloudlinux.com) (gcc version 4.4.7 20120313 (Red Hat 4.4.7-17) (GCC) ) #1 SMP Sun Jul 17 09:01:31 EDT 2016

**Client OS:** OS X El Capitan
**Browsers:** Chrome (), Firefox (47.0.1), Safari

**Comment by oshmyheliuk** [comment]
Still, can't  reproduce on fresh installed 2.0.4 and upgraded to 2.1.1 magento.
Maybe you have installed custom modules or specific settings.
Please, try to disable custom modules(if have), and reproduce this issue.

**Comment by milkolori** [comment]
Hi,

I disable all custom plugins but there was no success. I do have a second environment (my test environment, in my local computer) where I made a fresh installation of magento 2.1.1, and there everything works as expected.
I did some more investigation and I found that the problem is actually caused by the cache of the browser. When I disabled the cache using Chrome developer console then everithing works correctly. However in my test environment the cache does not cause any problem. Between both environments I see a difference in the requests  - in production environment the header "Pragma:no-cache" is missing.
